tests/test_dataflow/core/test_type_inference.py

```diff
@@ -0,0 +1,89 @@
+from typing import Dict
+
+import pytest
+
+from dataflow.core.definition import Definition
+from dataflow.core.lispress import lispress_to_program, parse_lispress
+from dataflow.core.program import TypeName
+from dataflow.core.type_inference import TypeInferenceError, infer_types
+
+
+def _do_inference_test(expr: str, expected: str, library: Dict[str, Definition]):
+    lispress = parse_lispress(expr)
+    program, _ = lispress_to_program(lispress, 0)
+    res = infer_types(program, library)
+    (expected_program, unused_idx) = lispress_to_program(parse_lispress(expected), 0,)
```

Comment:
`unused_idx` can be `_`?
Also remove the extra comma
```suggestion
    (expected_program, _) = lispress_to_program(parse_lispress(expected), 0)
```